src/BenchmarkDotNet/Toolchains/MonoWasm/WasmGenerator.cs

```diff
@@ -1,27 +1,82 @@
 ﻿using System.IO;
 using System.Text;
+using BenchmarkDotNet.Environments;
 using BenchmarkDotNet.Extensions;
 using BenchmarkDotNet.Helpers;
 using BenchmarkDotNet.Loggers;
 using BenchmarkDotNet.Running;
 using BenchmarkDotNet.Toolchains.CsProj;
 
+
 namespace BenchmarkDotNet.Toolchains.MonoWasm
 {
     public class WasmGenerator : CsProjGenerator
     {
         private readonly string CustomRuntimePack;
+        //TODO: Set this properly
+        private readonly bool Aot = true;
 
-        public WasmGenerator(string targetFrameworkMoniker, string cliPath, string packagesPath, string customRuntimePack)
+        public WasmGenerator(string targetFrameworkMoniker, string cliPath, string packagesPath, string customRuntimePack, bool aot)
             : base(targetFrameworkMoniker, cliPath, packagesPath, runtimeFrameworkVersion: null)
         {
+            Aot = aot;
             CustomRuntimePack = customRuntimePack;
         }
 
+
         protected override void GenerateProject(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
         {
+            WasmRuntime runtime = (WasmRuntime)buildPartition.Runtime;
+
+            if (runtime.Aot)
+            {
+                GenerateProjectAot(buildPartition, artifactsPaths, logger);
+            }
+            else
+            {
+                GenerateProjectInterpreter(buildPartition, artifactsPaths, logger);
+            }
+
+        }
+
+        protected  void GenerateProjectAot(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
+        {
+            BenchmarkCase benchmark = buildPartition.RepresentativeBenchmarkCase;
+            var projectFile = GetProjectFilePath(benchmark.Descriptor.Type, logger);
+
+            WasmRuntime runtime = (WasmRuntime)buildPartition.Runtime;
+
+            using (var file = new StreamReader(File.OpenRead(projectFile.FullName)))
+            {
+                var (customProperties, sdkName) = GetSettingsThatNeedsToBeCopied(file, projectFile);
+
+                string content = new StringBuilder(ResourceHelper.LoadTemplate("WasmAotCsProj.txt"))
+                    .Replace("$PLATFORM$", buildPartition.Platform.ToConfig())
+                    .Replace("$CODEFILENAME$", Path.GetFileName(artifactsPaths.ProgramCodePath))
+                    .Replace("$CSPROJPATH$", projectFile.FullName)
+                    .Replace("$TFM$", TargetFrameworkMoniker)
+                    .Replace("$PROGRAMNAME$", artifactsPaths.ProgramName)
+                    .Replace("$RUNTIMESRCDIR$", runtime.RuntimeSrcDir.ToString())
+                    .Replace("$COPIEDSETTINGS$", customProperties)
+                    .Replace("$CONFIGURATIONNAME$", buildPartition.BuildConfiguration)
+                    .Replace("$RUNTIMECONFIG$", "Debug")
```

Comment:
Why `Debug` and not `buildPartition.BuildConfiguration`?

```suggestion
                    .Replace("$RUNTIMECONFIG$", buildPartition.BuildConfiguration)
```

Reply:
This should actually be the configuration the runtime is built with, which is not necessarily the same as the build configuration. I have instead added an option for it.

Reply:
Debug builds contain zero optimizations enabled and we are rather not interested in their performance. Moreover, BDN produces an error for every dependency that was built without optimizations enabled. Please hardcode it to "Release" and remove the config switch.

We already have few dozens of console line args, I just try to not add more of not needed. Thanks!